browser/extensions/api/brave_rewards_api.cc

```diff
@@ -69,11 +70,96 @@ ExtensionFunction::ResponseAction BraveRewardsTipSiteFunction::Run() {
     return RespondNow(Error(tabs_constants::kTabNotFoundError,
                             base::IntToString(params->tab_id)));
   }
-  ::brave_rewards::OpenTipDialog(contents, params->publisher_key);
+
+  auto params_dict = std::make_unique<base::DictionaryValue>();
+  params_dict->SetString("publisherKey", params->publisher_key);
+  ::brave_rewards::OpenTipDialog(contents, std::move(params_dict));
 
   return RespondNow(NoArguments());
 }
 
+BraveRewardsTipTwitterUserFunction::BraveRewardsTipTwitterUserFunction()
+    : weak_factory_(this) {
+}
+
+BraveRewardsTipTwitterUserFunction::~BraveRewardsTipTwitterUserFunction() {
+}
+
+ExtensionFunction::ResponseAction
+BraveRewardsTipTwitterUserFunction::Run() {
+  std::unique_ptr<brave_rewards::TipTwitterUser::Params> params(
+      brave_rewards::TipTwitterUser::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  // Sanity check: don't allow tips in private / tor contexts,
+  // although the command should not have been enabled in the first place.
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  if (profile->IsOffTheRecord()) {
+    return RespondNow(
+        Error("Cannot tip Twitter user in a private context"));
+  }
+
+  auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
+  if (rewards_service) {
+    AddRef();
+    std::map<std::string, std::string> args;
+    args["user_id"] = params->tweet_meta_data.user_id;
+    args["name"] = params->tweet_meta_data.name;
+    args["screen_name"] = params->tweet_meta_data.screen_name;
+    rewards_service->SaveTwitterPublisherInfo(
+        args,
+        base::Bind(&BraveRewardsTipTwitterUserFunction::
+                   OnTwitterPublisherInfoSaved,
+                   weak_factory_.GetWeakPtr()));
+  }
+
+  return RespondNow(NoArguments());
+}
+
+void BraveRewardsTipTwitterUserFunction::OnTwitterPublisherInfoSaved(
+    std::unique_ptr<::brave_rewards::ContentSite> publisher_info) {
+  std::unique_ptr<brave_rewards::TipTwitterUser::Params> params(
+      brave_rewards::TipTwitterUser::Params::Create(*args_));
+
+  if (!publisher_info) {
+    // TODO(nejczdovc): what should we do in this case?
+    Release();
+    return;
+  }
+
+  // Get web contents for this tab
+  content::WebContents* contents = nullptr;
+  if (!ExtensionTabUtil::GetTabById(
+        params->tab_id,
+        Profile::FromBrowserContext(browser_context()),
+        include_incognito_information(),
```

Comment:
any harm in just setting this to `false` since we don't use enable tipping for incognito webcontents?

Reply:
Sounds fine to me, any objections there @NejcZdovc?

Reply:
no objections, tipping should be disabled in incognito

Reply:
Fixed.